addons/website_sale/controllers/main.py

```diff
@@ -561,12 +561,13 @@ def remove_product_image(self, image_res_model, image_res_id):
             raise NotFound()
 
         image_res_id = int(image_res_id)
-        if image_res_model == 'product.product':
-            request.env['product.product'].browse(image_res_id).write({'image_1920': False})
-        elif image_res_model == 'product.template':
-            request.env['product.template'].browse(image_res_id).write({'image_1920': False})
-        else:
+        if image_res_model == 'product.image':
             request.env['product.image'].browse(image_res_id).unlink()
+        elif image_res_model == 'product.product' and request.env['product.product'].browse(image_res_id)['image_variant_1920']:
+            # In the case of a suppression of an image of a product that has a
+            # field of type 'image_variant_1920', set this field to 'False' as
+            # it will not be updated at the save request.
+            request.env['product.product'].browse(image_res_id).write({'image_variant_1920': False})
```

Comment:
What about not calling this controller at all when trying to remove the main image (that was I was trying to say in my previous message) ? :wink:

Reply:
I'll need some more convincing, not sure to understand :grimacing: All I am asking is to delay this emptying of the image_variant_1920 field at the save lavel rather than with `remove-image` as a "pre-saving"... surely that should be possible :grimacing:

Reply:
@loco-odoo That's exactly my point: what is specific behavior should stay as specific code and what is generic should be implemented generically:
- Removing a product.image record -> that's a very specific feature (there is whole model dedicated to extra product images and records of those models have to be deleted/added to remove/add image via the editor) => this should stay as a RPC done on click of the button, then we have to save the whole page so that the page reloads with the proper rendering.
- Emptying the main product image -> if we want that feature (**), this should be a generic behavior: we have an image t-field that we want to empty. The editor could be in charge of "emptying that field" = removing the image / displaying the 404 image and the saving (`save_embedded_field`) could be in charge of actually save that "emptying of a field".

And even if we don't agree on that part: it is technically more consistent and "more efficient" if it is done that way:
- You save the page, the saving of the edited image is done at the same time, same as all other potentially edited t-field and other content in the page. + If the saving fails, then the saving fails: you don't have a half-save that has been done before the main save crashed.
- You avoid a RPC to be done in the case of removing the main product image (since you just let the normal RPC save to be done for that). That argument alone should be enough :wink: 

> I would just hide the remove button when it concerns the main image, I don't see the point in removing the main image. That would need PO validation though (there is a new PO, don't remember its quadrigram).

** Also, this :point_up: is still a possibility to me.

Reply:
> But isn't it consider as mixing specific code with generic code (as we would refer to product.product at the web_editor level)?

Exactly, that's why I did not imagine doing that but instead doing it generically :wink: In `save_embedded_field`, I imagined something like "if value/image is null, then the given field of the given model (you don't have to know it is product.product and image_variant_1920) is set to False".

Then a write of image_1920 to False should do the proper thing, whatever they are (if it is a computed field like you said (`_compute_image_1920`), then there could be an inverse function which does the proper things when written?).

> However, may I suggest also an other way of doing? [...] Sure, we can also see with the PO of ecommerce what he/she thinks about removing the possibility to remove a main image +1

Yes, at this point I think we really should have @AntoineVDV and its PO's opinion... :wink: